File: Lecture4/Ex45_CountingSpaces.java
import java.util.Scanner;

public class Ex45_CountingSpaces {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.print( "Enter a line : ");
        String s = input.nextLine();

        int num_spaces = 0;

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == ' ' ) num_spaces++;
        }

        System.out.println( "Num of spaces : " + num_spaces );
    }
}

File: Lecture4/Ex45_CountingSpaces.c
#include <stdio.h>
#define MAX_LEN 256

int main(void) {

    char buffer[MAX_LEN], *result;

    printf("Enter a line : ");
    result = fgets(buffer, MAX_LEN, stdin);

    int num_spaces = 0;

    for (int i = 0; result[i] != '\0'; i++) {
        if ( result[i] == ' ' ) num_spaces++;
    }

    printf("Number of spaces : %d\n", num_spaces);

    return 0;
}
